packages/adblocker/src/lists.ts

```diff
@@ -189,38 +202,113 @@ export function parseFilters(
       const filter = NetworkFilter.parse(line, config.debug);
       if (filter !== null) {
         networkFilters.push(filter);
+        if (preprocessorStack.length) {
+          filterStack.push(filter);
+        }
       }
     } else if (filterType === FilterType.COSMETIC && config.loadCosmeticFilters === true) {
       const filter = CosmeticFilter.parse(line, config.debug);
       if (filter !== null) {
         if (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false) {
           cosmeticFilters.push(filter);
+          if (preprocessorStack.length) {
+            filterStack.push(filter);
+          }
+        }
+      }
+    } else if (config.loadPreprocessors) {
+      const preprocessorType = detectPreprocessor(line);
+
+      if (preprocessorType === PreprocessorTypes.BEGIF) {
+        preprocessorStack.push({
+          start: filterStack.length,
+          condition: Preprocessor.getCondition(line),
+        });
+      } else if (preprocessorType !== PreprocessorTypes.INVALID && preprocessorStack.length > 0) {
+        const condition =
+          preprocessorStack.length === 1
+            ? preprocessorStack[0].condition
+            : preprocessorStack.map((entry) => `(${entry.condition})`).join('&&');
+
+        const existingPreprocessor = preprocessors.find(
+          (preprocessor) => preprocessor.condition === condition,
+        );
+
+        if (existingPreprocessor) {
+          for (
+            let i = filterStack.length - 1;
+            i >= preprocessorStack[preprocessorStack.length - 1].start;
+            i--
+          ) {
+            existingPreprocessor.filterIDs.add(filterStack.pop()!.getId());
+          }
+        } else {
+          const filterIDs: Set<number> = new Set();
+
+          for (
+            let i = filterStack.length - 1;
+            i >= preprocessorStack[preprocessorStack.length - 1].start;
+            i--
+          ) {
+            filterIDs.add(filterStack.pop()!.getId());
+          }
+
+          preprocessors.push(Preprocessor.parse(condition, filterIDs));
+        }
+
+        const last = preprocessorStack.pop();
+
+        if (preprocessorType === PreprocessorTypes.ELSE) {
+          preprocessorStack.push({
+            start: filterStack.length,
+            condition: `!(${last!.condition})`,
+          });
         }
       }
     }
   }
 
-  return { networkFilters, cosmeticFilters };
+  return { networkFilters, cosmeticFilters, preprocessors };
 }
 
-function getFilters(list: string, config?: Partial<Config>): (NetworkFilter | CosmeticFilter)[] {
-  const { networkFilters, cosmeticFilters } = parseFilters(list, config);
+function getFilters(
```

Comment:
I don't think we have to change the `getFilters` and maybe we should not as preprocessors are optional.